frontend/src/core/cells/cells.ts

```diff
@@ -217,10 +219,16 @@ const {
     const insertionIndex = before ? index : index + 1;
 
     let cellContents = code;
-    const cellEditorView = getCellEditorView(cellId as CellId);
-    if (cellEditorView) {
-      cellContents = extractHighlightedCode(cellEditorView);
-      state.cellData[cellId as CellId].edited = true;
+
+    if (includeSelectionAsInitialCode) {
+      const id = cellId as CellId;
+      const cellEditorView = getCellEditorView(id);
+      if (cellEditorView) {
+        const [highlighted, leftover] = extractHighlightedCode(cellEditorView);
+        cellContents = highlighted;
```

Comment:
we should only do this this if `highlighted` is not empty

Reply:
i think it will catch small bugs like `state.cellData[id].edited=true` being set incorrectly. 

also, doing `state.cellData[id].edited` is mutable, we will need to be immutable when using `react`. 

```python
state.cellData = {
   ...state.cellData,
   [id] : {
     ...state.cellData[id],
     code: ...,
     edited: ...
   }
}
```